pkg/apis/v1beta1/nodepool.go

```diff
@@ -202,3 +210,72 @@ func (pl *NodePoolList) OrderByWeight() {
+	vals := make([]intstr.IntOrString, len(in.Spec.Disruption.Budgets))
+	errs := make([]error, len(in.Spec.Disruption.Budgets))
+	workqueue.ParallelizeUntil(ctx, len(in.Spec.Disruption.Budgets), len(in.Spec.Disruption.Budgets), func(i int) {
+		val, err := in.Spec.Disruption.Budgets[i].GetAllowedDisruptions(c)
+		if err != nil {
+			errs[i] = fmt.Errorf("invalid budget %s, %w", lo.FromPtr(in.Spec.Disruption.Budgets[i].Crontab), err)
+		}
+		vals[i] = val
+	})
+	if err := multierr.Combine(errs...); err != nil {
+		return intstr.IntOrString{}, intstr.IntOrString{}, err
+	}
+	minIntVal, minPercentVal := math.MaxInt64, math.MaxInt64
+	for i := range vals {
+		val := vals[i]
+		// The crontab wasn't active
+		if val.IntVal == -1 {
+			continue
+		}
+		// This returns the percent value if it's a string, and the raw value if it's an int.
+		temp, err := intstr.GetScaledValueFromIntOrPercent(lo.ToPtr(val), 100, false)
+		if err != nil {
+			// Should almost never happen since this is validated when the nodepool is applied
+			return intstr.IntOrString{}, intstr.IntOrString{}, fmt.Errorf("getting intstr scaled value, %w", err)
+		}
+		if val.Type == intstr.Int {
+			minIntVal = lo.Ternary(temp < minIntVal, temp, minIntVal)
+		} else {
+			minPercentVal = lo.Ternary(temp < minPercentVal, temp, minPercentVal)
+		}
+	}
+	// return the values, defaulting to -1 if the value is MaxInt64
+	return intstr.FromInt(lo.Ternary(minIntVal == math.MaxInt64, -1, minIntVal)), intstr.FromString(fmt.Sprintf("%d%%", lo.Ternary(minPercentVal == math.MaxInt64, -1, minPercentVal))), nil
+}
+
+// GetAllowedDisruptions returns an intstr.IntOrString that can be used a comparison
+// for calculating if a disruption action is allowed. It returns an error if the
+// crontab is invalid. This returns -1 if the value is unbounded.
+func (b *Budget) GetAllowedDisruptions(c clock.Clock) (intstr.IntOrString, error) {
+	active, err := b.IsActive(c)
+	if err != nil {
+		return intstr.IntOrString{}, err
+	}
+	return lo.Ternary(active, b.MaxUnavailable, intstr.FromInt(-1)), nil
+}
+
+// IsActive takes a clock as input and returns if a budget is active.
+// It walks back in time the time.Duration associated with the crontab,
+// and checks if the next time the schedule will hit is before the current time.
+// If the last crontab hit is exactly the duration in the past, this means the
+// schedule is active, as any more crontab hits in between would only extend this
+// window. This ensures that any previous crontab hits for a schedule are considered.
+func (b *Budget) IsActive(c clock.Clock) (bool, error) {
+	if b.Crontab == nil && b.Duration == nil {
+		return true, nil
+	}
+	schedule, err := cron.ParseStandard(lo.FromPtr(b.Crontab))
+	if err != nil {
+		return false, fmt.Errorf("parsing crontab, %w", err)
+	}
+	// Walk back in time for the duration associated with the crontab
+	checkPoint := c.Now().Add(-lo.FromPtr(b.Duration).Duration)
+	nextHit := schedule.Next(checkPoint)
```

Comment:
Is there a clearer way to do this logic? I get how it's working but the way it's written makes it a little hard to follow. Perhaps an example in the comment could help clear it up but I'm also curious if there's a concept of "between" where we can get the last crontab and see if we are currently between that time and the duration after it

Reply:
If you have a suggestion on how to make it clearer in code, I'm open to it :D 

Let me know if this example helps:
```
Crontab = @hourly == 0 * * * * and time.Now() = 10:15 and duration = 30m, we'll walk back to 9:45, get the next crontab hit from there, which should be 10, which is before 10:15, meaning the cron is active. 
Doing this with the same but time.Now() = 10:45, we walk back in time to 10:15 and then get the next crontab hit, which is 11:00. Since this is after 10:45, we say that the cron is not active.
```

Reply:
I think (maybe thats just me) the naming can be a bit confusing. Rather than calling it `nextHit` maybe it should be something like `closestLastHit`?

Reply:
I think its difficult given the cron parser only gives you `Next` based on a time and finding the last crontab is quite hard to do. I think this method make sense even though is slightly complicated to understand